plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -7623,6 +7624,95 @@ public void testSystemTables()
         assertQueryFails("TABLE \"nation$foo\"", "\\Qline 1:1: Table '%s.%s.\"nation$foo\"' does not exist".formatted(catalog, schema));
     }
 
+    @Test
+    public void testExtraProperties()
+    {
+        String tableName = "create_table_with_multiple_extra_properties_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.one', 'extra.property.two'], ARRAY['one', 'two']))".formatted(tableName));
+
+        assertQuery(
+                "SELECT value FROM \"%s$properties\" where key in ('extra.property.one', 'extra.property.two')".formatted(tableName),
+                "VALUES ('one'), ('two')");
+
+        assertUpdate("DROP TABLE %s".formatted(tableName));
+    }
+
+    @Test
+    public void testReplaceTableExtraProperties()
+    {
+        String tableName = "replace_table_with_multiple_extra_properties_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.one', 'extra.property.two'], ARRAY['one', 'two']))".formatted(tableName));
+        assertUpdate("CREATE OR REPLACE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.three'], ARRAY['three']))".formatted(tableName));
+
```

Comment:
let's verify that `'extra.property.one', 'extra.property.two'` are not present anymore in the properties for  the current snapshot of the table.